[Design] 공통 컴포넌트인 기본 TextField를 추가합니다.

Rookie0031 opened the pull request:
## 관련 이슈
- close #24 

## 배경
회원가입 및 밴드 생성, 합주곡 추가 등에 사용되는 공통 컴포넌트 TextField UI를 구현합니다.

## 작업 내용
#### 1. 공통 컴포넌트 BasicTextField 클래스 추가
#### 2. (1)번 사항 구현에 필요한 TextField Extension 추가
#### 3. BaseViewController에 일부 필요 코드 추가

### BasicTextField 
 일반적인 UI 컴포넌트처럼 인스턴스화해서 사용하시면 되고 init할 때 플레이스홀더를 입력해줘야합니다.
기본적으로 기기 사이즈의 0.9, 그리고 고정 높이 55를 가집니다. 필요한 경우 각자의 ViewController에서 크기와 위치를 잡아줄 수 있고 UIStackView에 사용할 경우 StackView의 크기만 설정해주고 arrangedSubView에 넣어주면 됩니다. 다음은 예시 사용법입니다.


    class LandingViewController: BaseViewController {
        
        let testTextField = BasicTextField(placeholder: "테스트 용입니다")
        
        lazy var testStackView = UIStackView(arrangedSubviews: [testTextField])
        
        override func viewDidLoad() {
            super.viewDidLoad()
            
            self.view.addSubview(testStackView)
            
            testStackView.axis = .vertical
            testStackView.constraint(leading: self.view.leadingAnchor, trailing: self.view.trailingAnchor, padding: UIEdgeInsets(top: 0, left: 25, bottom: 0, right: 25))
            testStackView.constraint(centerX: self.view.centerXAnchor, centerY: self.view.centerYAnchor)
        }
    }

### UITextField+Extension
 TextField 기본 형태를 만들 수 있는 코드가 익스텐션으로 추가되었습니다. 기본적으로 플레이스홀더를 입력하며 필요한 경우 최대 글자수를 파라미터로 줄 수 있습니다. 다음 PR에 올릴 글자수 제한 TextField에서도 사용될 예정입니다.
 UITextField에는 왼쪽과 오른쪽에 작은 View를 넣을 수 있도록 leftView와 rightView라는 프로퍼티가 있습니다. 이를 이용해서 텍스트 필드의 맨 앞에 패딩값을 주는 효과를 위해 leftView에는 EmptyView가 추가되었습니다.

### BaseViewController
 BaseViewController에 빈화면을 터치하면 키보드가 내려가는 코드와 배경색상 설정 코드가 추가되었습니다. 필요하다면 상속해서 사용하시면 됩니다.

    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        view.endEditing(true)
    }

### 기타
DeviceSize를 좀 더 쉽게 가져올 수 있도록 다음 코드가 선언되었습니다.

    struct DeviceSize {
        static var width = UIScreen.main.bounds.width
        static var height = UIScreen.main.bounds.height
    }

## 리뷰 노트

 다른분들도 쉽게 쓸 수 있도록 공통 컴포넌트 구성을 고민하긴 했는데 고민이 너무 많아지다보니 진행이 더뎌지는 거 같아 일단 PR을 올립니다 그에따라  불필요한 코드는 없는지 의문이 들긴합니다. 컴포넌트에 기본적인 사이즈를 설정해놓은것이 큰 문제없는 코드인지 궁금하네요.

## 테스트 방법
#### ❗️공용 컴포넌트를 테스트할 방법이 딱히 없는데, 현재 만들어진 LandingViewController에 테스트 코드를 추가하는 방식으로 진행하는 것을 제안합니다.

#### LandViewController에 // MARK: 기본 TextField 실험 코드 이하 코드를 주석처리를 풀고 진행해주시면 됩니다. 

## 스크린샷
<img width="300" alt="스크린샷1" src="https://user-images.githubusercontent.com/103009135/213960972-b8f3c3e2-d555-4837-98fe-5709830f06e9.png">

compuTasha committed:
Initial commit

compuTasha committed:
[Setting] 이슈 및 PR 템플릿 추가 (#1)

compuTasha committed:
[Setting] PR 템플릿 수정 (#1)

compuTasha committed:
Merge pull request #2 from extreme-rock/feature/1-set-issue-templates

[Setting] 이슈 및 PR 템플릿을 추가했습니다.

GODNOEL committed:
[Setting] .gitignore 수정 (#4)

SwiftPackageManager를 추가 시 모든 .xcodeproj 파일도 무시하게되어 해당 항목을 주석처리로 제외시켜 주었습니다.
(245 line)

compuTasha committed:
[Add] Xcode 프로젝트 생성 (#3)

compuTasha committed:
[Chore] 미니멈 타겟을 15.0 으로 설정 (#3)

compuTasha committed:
[Chore] iPad를 타겟에서 제외 (#3)

compuTasha committed:
[Chore] 가로모드 방지 (#3)

compuTasha committed:
[Chore] 들여쓰기 4칸으로 설정 (#3)

compuTasha committed:
[Setting] 프로젝트 폴더링 (#3)

GODNOEL committed:
Merge pull request #5 from extreme-rock/feature/4-set-gitignore

[Setting] 프로젝트에 적합하게 .gitignore 을 수정했습니다.

GODNOEL committed:
[Setting] auto_assign.yml 파일 생성 (#7)

compuTasha committed:
[Chore] info.plist 'StoryBoard Name' 삭제 (#3)

compuTasha committed:
[Add] LandingViewController를 시작 뷰컨으로 설정 (#3)

GODNOEL committed:
Merge pull request #8 from extreme-rock/feature/7-set-auto-assign

[Setting] PR의 reviewer를 자동 할당을 위해 auto-assign을 설정합니다.

compuTasha committed:
Merge pull request #6 from extreme-rock/feature/3-init-xcode-project

[Setting] Xcode 프로젝트를 생성하고 초기 세팅을 했습니다.

ccdkk committed:
[Setting] Constraint 관련 extension을 추가합니다. (#10)

compuTasha committed:
[Setting] CocoaPods 추가 (#9)

compuTasha committed:
[Setting] podfile platform 버전 변경 (#9)

9.0 -> 15.0

compuTasha committed:
[Chore] AppDelegate에 구글맵 API 키 영역 추가 (#9)

GODNOEL committed:
clear git cache

ccdkk committed:
Merge branch 'develop' of https://github.com/extreme-rock/GetARock-iOS into feature/10-setting-constraint

ccdkk committed:
Merge pull request #11 from extreme-rock/feature/10-setting-constraint

[Setting] Constraint 관련 extension을 추가합니다.

compuTasha committed:
Merge branch 'develop' of feature/9-set-cocoapods

compuTasha committed:
Merge pull request #12 from extreme-rock/feature/9-set-cocoapods

[Setting] CocoaPods을 설치하고 GoogleMaps를 추가합니다.

compuTasha committed:
[Add] MapsInfo.plist 파일 추가 (#13)

compuTasha committed:
[Add] MapsInfo.plist에 접근하기 위한 Bundle 추가 (#13)

ccdkk committed:
[Setting] xcode build test (#14)

ccdkk committed:
[Setting] os 버전 latest로 변경 (#14)

GODNOEL committed:
[Fix].gitignore 적용을 위해 깃 캐쉬를 삭제 (#15)

GODNOEL committed:
[Design] Color hex 값 추가 (#25)

compuTasha committed:
[Rename] Bundle extension 파일명 수정 (#13)

GetARock++Bundle -> Bundle+Extension

ccdkk committed:
Merge pull request #21 from extreme-rock/feature/14-setting-action

[Setting] GitHub Action에 build test를 추가합니다.

GODNOEL committed:
[Style] 변수명 오타 수정 (#25)

GODNOEL committed:
[Design] font type 추가 (#30)

GODNOEL committed:
Revert "[Style] 변수명 오타 수정 (#25)"

This reverts commit 84a4ee0d916a1df116bee5280bed7bc94a705332.

GODNOEL committed:
[Style] 변수명 오타 수정 (#25)

compuTasha committed:
Merge branch 'feature/13-add-api-key' of https://github.com/extreme-rock/GetARock-iOS into feature/20-design-map-ui

compuTasha committed:
[Add] MainMapViewController 추가 (#20)

compuTasha committed:
[Design] 지도 다크모드 스타일 적용 (#20)

compuTasha committed:
[Resource] 메인 맵 버튼 이미지 추가 (#20)

compuTasha committed:
[Style] MARK 주석 추가 (#20)

compuTasha committed:
[Design] 메인 맵 하단 버튼 추가 (#20)

compuTasha committed:
Merge pull request #17 from extreme-rock/feature/13-add-api-key

[Add] Google Maps API KEY를 github 상에서 숨깁니다.

compuTasha committed:
[Design] 메인 맵 상단 버튼 추가 (#20)

compuTasha committed:
[Design] 메인 맵 현 위치 라벨 추가 (#20)

compuTasha committed:
[Style] 개행 수정 (#20)

GODNOEL committed:
[Style] 주석 오타 수정 (#25)

compuTasha committed:
[Refactor] 메인 맵 레이아웃 코드 리팩토링 (#20)

compuTasha committed:
[Rename] 현위치로 이동하는 버튼 네이밍 수정 (#20)

currentLocationButton -> moveToCurrentLocationButton

GODNOEL committed:
Merge pull request #29 from extreme-rock/feature/25-extension-color

[Design] Extension Color 값들을 추가합니다.

GODNOEL committed:
[Add] 폰트 추가 및 변수명 변경 (#30)

GODNOEL committed:
Merge branch '30-extension' into 'develop'

GODNOEL committed:
Merge branch 'develop' of https://github.com/extreme-rock/GetARock-iOS into feature/15-delete-gitcache

GODNOEL committed:
Merge pull request #22 from extreme-rock/feature/15-delete-gitcache

[Fix] .gitignore 적용을 위해 이전 깃 캐쉬를 삭제를 재진행하였습니다

GODNOEL committed:
Merge pull request #31 from extreme-rock/feature/30-extension-font

[Design] extension UIFont를 통해 font Type를 추가합니다.

compuTasha committed:
[Style] 개행 수정 (#20)

compuTasha committed:
[Design] padding 값 변경 (#20)

compuTasha committed:
Merge branch 'develop' of https://github.com/extreme-rock/GetARock-iOS into feature/20-design-map-ui

compuTasha committed:
[Rename] Screens > View 폴더 삭제 (#20)

compuTasha committed:
[Design] UIFont extension 적용 (#20)

compuTasha committed:
Merge pull request #35 from extreme-rock/feature/20-design-map-ui

[Design] 메인 지도 뷰 UI를 구성합니다.

Rookie0031 committed:
[Design] 기본 TextField 구현을 위한 extension 및 class 추가

Rookie0031 committed:
[Design] 기본 TextField 구현을 위한 extension 및 class 추가

Rookie0031 committed:
[Design] 기본 TextField 관련 코드 추가

Rookie0031 committed:
[Design] 기본 TextField 관련 코드 추가

Rookie0031 committed:
[Feat] BaseController에 빈 화면 터치시 키보드 숨김 처리 추가

Rookie0031 committed:
[Feat] BaseController에 빈 화면 터치시 키보드 숨김 처리 추가

Rookie0031 committed:
[Add] 불필요한 파일 삭제 및 주석 작업

Rookie0031 committed:
[Add] 불필요한 파일 삭제 및 주석 작업

compuTasha commented on GetARock/GetARock/Global/Base/BaseViewController.swift:
```diff
@@ -8,19 +8,17 @@
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        // Get the new view controller using segue.destination.
-        // Pass the selected object to the new view controller.
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        view.endEditing(true)
+    }
+    
+    private func attribute() {
+        view.backgroundColor = .dark01
     }
```
p3;

코드 컨벤션에 따라 self를 명시적으로 사용하는 게 좋을 것 같습니다!

compuTasha replied:
대부분 뷰의 배경 색상이 dark01이고,
앱 전반적으로 텍스트 필드도 많아서 둘 다 BaseViewController에 선언해두면 편할 것 같네요! 쏘 굿 👍

compuTasha commented on GetARock/GetARock/Global/Base/BaseViewController.swift:
```diff
@@ -8,19 +8,17 @@
     
-    /*
-    // MARK: - Navigation
-
-    // In a storyboard-based application, you will often want to do a little preparation before navigation
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        // Get the new view controller using segue.destination.
-        // Pass the selected object to the new view controller.
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        view.endEditing(true)
+    }
+    
+    private func attribute() {
+        view.backgroundColor = .dark01
     }
```
p3;

코드 컨벤션에 따라 self를 명시적으로 사용하는 게 좋을 것 같습니다!

Rookie0031 replied:
오 감사합니다! 반영완료~ https://github.com/extreme-rock/GetARock-iOS/pull/37/commits/43247341b7c06d57531e5a3c175be25d219b239c

compuTasha commented on GetARock/GetARock/Global/Utils/DeviceSize.swift:
```diff
@@ -0,0 +1,13 @@
+//
+import UIKit
+
+struct DeviceSize {
+    static var width = UIScreen.main.bounds.width
+    static var height = UIScreen.main.bounds.height
+}
```
p2;

스크린 크기는 static let 으로 선언하는 게 어떨까요? 👀

Rookie0031 replied:
감사합니다 반영 완료 ! https://github.com/extreme-rock/GetARock-iOS/pull/37/commits/43247341b7c06d57531e5a3c175be25d219b239c

Rookie0031 committed:
[Add] 코드리뷰 사항 반영

Rookie0031 committed:
[Add] 코드리뷰 사항 반영

compuTasha commented on GetARock/GetARock.xcodeproj/project.pbxproj:
```diff
@@ -690,6 +702,7 @@
 				ASSETCATALOG_COMPILER_GLOBAL_ACCENT_COLOR_NAME = AccentColor;
 				CODE_SIGN_STYLE = Automatic;
 				CURRENT_PROJECT_VERSION = 1;
+				DEVELOPMENT_TEAM = YYP248S73Y;
```
p1;

루키 Team ID가 따라왔어요..! 가능하면 로그까지 삭제하는 게 좋을 것 같아요..!

참고
- https://www.whatwant.com/entry/Git-특정-파일에-대한-이력-삭제
- https://velog.io/@jieboong_/Github-특정-파일-기록-삭제
- https://asecurity.dev/entry/Git-특정-파일-히스토리-삭제-하기

Rookie0031 replied:
헉 감사합니다! 근데 [git branch-filter](https://www.whatwant.com/entry/Git-특정-파일에-대한-이력-삭제) 이걸 진행하긴 했는데 제대로되었는지는 잘모르겟네요? 혼자하려다가 일이 커질 수 있어서 혹시 비슷한 경험하고 해결한적 있으신분 있으실까요?? @GODNOEL  @ccdkk  @Valentino1994

compuTasha commented on GetARock/GetARock/Global/Extension/UITextField+Extension.swift:
```diff
@@ -0,0 +1,60 @@
+    static func makeBasicTextField(placeholder: String, characterLimit: Int? = nil) -> UITextField {
+        let textField: UITextField = {
+            
+            $0.attributedPlaceholder = NSAttributedString(string: placeholder, attributes: [.foregroundColor: UIColor.gray02, .font: UIFont.setFont(.content)])
```
p3;

120자가 넘어가서 줄바꿈이 필요해보여요

```
            $0.attributedPlaceholder = NSAttributedString(
                string: placeholder,
                attributes: [.foregroundColor: UIColor.gray02, .font: UIFont.setFont(.content)]
            )
```

Rookie0031 replied:
감사합니다 반영완료 ~ https://github.com/extreme-rock/GetARock-iOS/pull/37/commits/71741576c799a5fd181ee88ff23d00578873a88d

compuTasha commented on GetARock/GetARock/Global/Extension/UITextField+Extension.swift:
```diff
@@ -0,0 +1,60 @@
+            
+            if let characterLimit { $0.maxCount = characterLimit }
+            
+            $0.layer.borderWidth = 2
```
p1;

피그마를 확인하니 border 두께가 1인 것 같습니다
수정 부탁드립니다

Rookie0031 replied:
앗 제가 놓쳤었군요 감사합니다!!! https://github.com/extreme-rock/GetARock-iOS/pull/37/commits/71741576c799a5fd181ee88ff23d00578873a88d

compuTasha commented on GetARock/GetARock/Global/Extension/UITextField+Extension.swift:
```diff
@@ -0,0 +1,60 @@
+    }
+}
+
+final class EmptyView: UIView {
```
p2;

emptyView는 보통 콘텐츠가 없을 때 보여주는 뷰(e.g. 빈 테이블, 빈 콜렉션뷰)에 쓰이는 네이밍인 것 같아서
`paddingView`와 같은 네이밍으로 변경하는 게 어떨까요?

Rookie0031 replied:
좋네요~ 좀더 정확하게 하기위해서 TextFieldPaddingView로 하였습니다! https://github.com/extreme-rock/GetARock-iOS/pull/37/commits/71741576c799a5fd181ee88ff23d00578873a88d

compuTasha commented on GetARock/GetARock/Global/UIComponent/BasicTextField.swift:
```diff
@@ -0,0 +1,32 @@
+        setupLayout()
+    }
+    
+    private func setupLayout() {
+        self.constraint(.widthAnchor, constant: DeviceSize.width * 0.9)
+        self.constraint(.heightAnchor, constant: 55)
+
```
p2;

`DeviceSize.width * 0.9` 나 `55`와 같은 매직 넘버를 
`textFieldWidth`, `textFieldHeight` 와 같은 상수로 만들어두고 관리하면 어떨까요?
추후 요구사항이 변경됨에 따라 수정해야 할 때 검색하기 쉬울 것 같습니다..

Rookie0031 replied:
좋습니다! 반영했습니다~ https://github.com/extreme-rock/GetARock-iOS/pull/37/commits/337c65297b040fce26c1ee3e66bf5424abe8ef59

compuTasha commented on GetARock/GetARock/Screens/Landing/LandingViewController.swift:
```diff
@@ -7,23 +7,26 @@
+        super.viewDidLoad()
+    }
+}
+
+// MARK: 기본 TextField 실험 코드
+//class LandingViewController: BaseViewController {
+//
+//    let testTextField = BasicTextField(placeholder: "테스트 용입니다")
+//
+//    lazy var testStackView = UIStackView(arrangedSubviews: [testTextField])
+//
+//    override func viewDidLoad() {
+//        super.viewDidLoad()
+//
+//        self.view.addSubview(testStackView)
+//
+//        testStackView.axis = .vertical
+//        testStackView.constraint(leading: self.view.leadingAnchor, trailing: self.view.trailingAnchor, padding: UIEdgeInsets(top: 0, left: 25, bottom: 0, right: 25))
+//        testStackView.constraint(centerX: self.view.centerXAnchor, centerY: self.view.centerYAnchor)
+//    }
+//}
```
테스트 코드 감사합니다! 덕분에 쉽게 확인했습니다👍
이부분은 머지 전에 원래 코드로 되돌린 후 머지 부탁드립니다!

compuTasha commented on GetARock/GetARock/Global/UIComponent/BasicTextField.swift:
```diff
@@ -0,0 +1,32 @@
+    init(placeholder: String) {
+        self.placeholder = placeholder
+        super.init(frame: .zero)
+        setupLayout()
```
p3;

self 무새 지나갑니다.. 
<img src="https://user-images.githubusercontent.com/51108729/214000404-1cd49e6f-af45-4b49-8841-fd810c71bf8e.gif" width="50">

Rookie0031 replied:
ㅋㅋㅋㅋㅋㅋㅋ역시 컨벤션의 제왕 감사합니다 👍!! https://github.com/extreme-rock/GetARock-iOS/pull/37/commits/cfd4737ce7f65931aed1a2c3d96f9d91a196233b

Rookie0031 committed:
[Style] 코드리뷰 사항 반영

1. 텍스트 필드 내 패딩값을 주기 위한 뷰의 네이밍 변경
2. 리뷰어 요청에 따른 일부 코드에서 개행
3. Development Team 커밋 사항 삭제

Rookie0031 committed:
[Style] 코드리뷰 사항 반영

1. 텍스트 필드 내 패딩값을 주기 위한 뷰의 네이밍 변경
2. 리뷰어 요청에 따른 일부 코드에서 개행
3. Development Team 커밋 사항 삭제

Rookie0031 committed:
[Style] 코드 리뷰 사항 반영

특정 함수 실행시 self를 명시함

Rookie0031 committed:
[Style] 코드 리뷰 사항 반영

특정 함수 실행시 self를 명시함

Rookie0031 committed:
Merge branch 'feature/24-add-basic-textField' of https://github.com/extreme-rock/GetARock-iOS into feature/24-add-basic-textField

Rookie0031 committed:
[Style] 기본 TextField 크기 관련 코드 네이밍 수정

Rookie0031 committed:
Merge branch 'feature/24-add-basic-textField' of https://github.com/extreme-rock/GetARock-iOS into feature/24-add-basic-textField

compuTasha commented:
p1;

루키 Team ID가 따라왔어요..! 가능하면 로그까지 삭제하는 게 좋을 것 같아요..!

참고
- https://www.whatwant.com/entry/Git-특정-파일에-대한-이력-삭제
- https://velog.io/@jieboong_/Github-특정-파일-기록-삭제
- https://asecurity.dev/entry/Git-특정-파일-히스토리-삭제-하기

--- 

헉 감사합니다! 근데 [git branch-filter](https://www.whatwant.com/entry/Git-%ED%8A%B9%EC%A0%95-%ED%8C%8C%EC%9D%BC%EC%97%90-%EB%8C%80%ED%95%9C-%EC%9D%B4%EB%A0%A5-%EC%82%AD%EC%A0%9C) 이걸 진행하긴 했는데 제대로되었는지는 잘모르겟네요? 혼자하려다가 일이 커질 수도 있을까봐 걱정되네요 ㅋㅋㅋ... 혹시 비슷한 경험하고 해결한적 있으신분 있으실까요?? @GODNOEL @ccdkk @Valentino1994

ccdkk commented on GetARock/GetARock/Global/UIComponent/BasicTextField.swift:
```diff
@@ -0,0 +1,32 @@
+//
+import UIKit
+
+final class BasicTextField: UIView {
+
+    var placeholder: String
+
+    private lazy var textField: UITextField = UITextField.makeBasicTextField(placeholder: placeholder)
+
+    init(placeholder: String) {
+        self.placeholder = placeholder
+        super.init(frame: .zero)
+        self.setupLayout()
+    }
+    
```
p2;
`makeBasicTextField`가 limit를 인자로 받는 것 같은데, BasicTextField에는 입력할 곳이 없는 것 같아요.

p2;
placeholder가 BasicTextField에서만 사용된다면 접근제어자 붙여주세요!

Rookie0031 replied:
BasicTextField에는 글자수 제한이 없기 때문에 init할 때 사용하지않게 처리했습니다!

접근제한자 감사합니다~ 반영했습니다 ! https://github.com/extreme-rock/GetARock-iOS/pull/37/commits/02c2e55c527f6d4008556a8e99faf2e09b007d20

ccdkk replied:
아하 BasicTextField는 글자수 제한이 없는 textField 군요!

ccdkk commented on GetARock/GetARock/Global/UIComponent/BasicTextField.swift:
```diff
@@ -0,0 +1,32 @@
+        self.constraint(.heightAnchor, constant: TextFieldSize.height)
+
+        self.addSubview(textField)
+        textField.constraint(top: self.topAnchor, leading: self.leadingAnchor, bottom: self.bottomAnchor, trailing: self.trailingAnchor)
```
p2;
`textField.constraint(to: self)`로 간단하게 나타낼 수 있을 것 같습니다!

ccdkk commented on GetARock/GetARock/Global/Extension/UITextField+Extension.swift:
```diff
@@ -0,0 +1,63 @@
+//
+import UIKit
+
+extension UITextField {
+    static func makeBasicTextField(placeholder: String, characterLimit: Int? = nil) -> UITextField {
+        let textField: UITextField = {
```
p5;
UITextField의 Extension으로 구현하신 이유가 궁급합니다. 
공통 컴포넌트를 만드는 이유가 여러 VC에서 `BasicTextField()`로 활용하기 위함이라고 생각하고, 그렇게 되면 `makeBasicTextField()`는 BasicTextField를 만들어줄 때만 사용될 것이라고 생각하여, extension에 넣기에 애매하다는 생각이 드는데 어떻게 생각하시나요?

Rookie0031 replied:
좋은 질문입니다!! 

음.. 사실 딱히 extension으로 빼야만하는 엄청난? 이유는 없습니다 ㅋㅋ.. 그냥 좀 코드를 잘게 나누고 가독성을 높이기 위함입니다!
다음 PR에는 글자수 제한과 텍스트 중복 검사 버튼을 포함하는 텍스트 필드인 TextLimitTextField 를 구현하는데 여기서 makeBasicTextField가 또 사용됩니다!

makeBasicTextField에서 기본적인 텍스트 필드 옵션들을 다 설정해주기 때문에 TextLimitTextField에서는 중복 검사 버튼의 위치를 잡는다거나 중복 검사 로직을 넣는 코드만 구성할 수 있다는 장점? 정도가 있는거 같습니다.

<img width="600" alt="image" src="https://user-images.githubusercontent.com/103009135/214319269-de95c909-d6ef-47cf-853e-3a1023a7a351.png">

ccdkk replied:
아 Basic과 limit 있는 textField가 구분되는군요~! 구체적인 설명 감사합니당

ccdkk reviewed (commented):
앞으로 사용될 소중한 공통 컴포넌트.. 만들어주셔서 감사합니다. 고생하셨어요!
테스트코드도 써주셔서 덕분에 쉽게 테스트할 수 있었습니다! 센스쟁이..시군요..

ccdkk commented:
> 앞으로 사용될 소중한 공통 컴포넌트.. 만들어주셔서 감사합니다. 고생하셨어요!
> 테스트코드도 너무 좋아요!

ccdkk commented:
@Rookie0031 
헉 실수로 pr을 닫아서 불필요한 로그가 남았네요 ㅜㅜ 죄송합니다..

GODNOEL commented on GetARock/GetARock/Global/Base/BaseViewController.swift:
```diff
@@ -8,19 +8,17 @@
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        // Get the new view controller using segue.destination.
-        // Pass the selected object to the new view controller.
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        self.view.endEditing(true)
+    }
```
오 바깥영역 터치하면 편집모드 끝나는 코드
`BaseViewController` 에 넣으신 거 센스 있으시네요!

GODNOEL commented on GetARock/GetARock/Global/Extension/UITextField+Extension.swift:
```diff
@@ -0,0 +1,63 @@
+        get {
+            MaxCount.number
+        }
+        set(newValue) {
+            MaxCount.number = newValue
+        }
```
Q
새로운 숫자가 들어가면 유효성검사 최대 숫자가 바뀌는 부분인가요?

Rookie0031 replied:
네 맞습니다~ UITextField Extension에 프로퍼티를 추가하고 싶은데 저장 프로퍼티는 익스텐션에 쓸 수 없어서 저렇게 get, set으로 연산프로퍼티로 구현해놨습니다!

GODNOEL reviewed (commented):
루키 텍스트필드 구현하느라 고생하셨어요!👍
다른분들이 리뷰 잘 달아주셔서 해당 리뷰들 해결되면 어프루드 드리겠습니다!!

Rookie0031 committed:
[Style] 코드 리뷰 사항 반영

compuTasha reviewed (approved):
스윽 어프루브 두고 갑니다^^

ccdkk reviewed (approved):
고생하셨습니다!
